Backpack/src/main/java/net/skyscanner/backpack/text/BpkText.kt

```diff
@@ -98,16 +102,41 @@ open class BpkText(
       weight = Weight.values()[weightArg]
     }
 
+    // Adding tint and compoundDrawables does not work. Converting compoundDrawables to compoundDrawablesRelative
+    if (this.layoutDirection == View.LAYOUT_DIRECTION_LTR) {
+      this.setCompoundDrawablesRelative(
+        compoundDrawablesRelative[0] ?: compoundDrawables[0],
+        compoundDrawablesRelative[1] ?: compoundDrawables[1],
+        compoundDrawablesRelative[2] ?: compoundDrawables[2],
+        compoundDrawablesRelative[3] ?: compoundDrawables[3])
+    } else {
+      this.setCompoundDrawablesRelative(
+        compoundDrawablesRelative[0] ?: compoundDrawables[2],
+        compoundDrawablesRelative[1] ?: compoundDrawables[1],
+        compoundDrawablesRelative[2] ?: compoundDrawables[0],
+        compoundDrawablesRelative[3] ?: compoundDrawables[3])
+    }
+    if (a.getColorStateList(R.styleable.BpkText_drawableTint) != null) {
+      val drawableTint = a.getColorStateList(R.styleable.BpkText_drawableTint)
+      setDrawableTint(drawableTint!!.getColorForState(EMPTY_STATE_SET, Color.WHITE))
```

Comment:
Why is the `getColorForState` call needed?

Reply:
Also is it possible to avoid non-null assertion (`!!`)?

Reply:
The color passed via `xml` is of the type `ColorStateList` while the `colorFilter` requires just one color, hence the conversion. Removed the `!!`